src/cpp/fastdds/subscriber/history/DataReaderHistory.cpp

```diff
@@ -86,9 +86,9 @@ DataReaderHistory::DataReaderHistory(
         resource_limited_qos_.max_samples_per_instance = std::numeric_limits<int32_t>::max();
     }
 
-    if (type_->m_isGetKeyDefined)
+    if (type_->is_compute_key_provided)
     {
-        get_key_object_ = type_->createData();
+        get_key_object_ = type_->create_data();
```

Comment:
With the new `compute_key` that takes a serialized payload, we can remove `get_key_object_`.

Reply:
Done in d0adb7824